docs/spec/staking/spec-technical.md

```diff
@@ -375,111 +389,262 @@ type TxUnbond struct {
 
 ```
 unbond(tx TxUnbond):
-
-	// get delegator bond
 	bond = loadDelegatorBond(store, sender, tx.PubKey)
 	if bond == nil then return 
-
-	// subtract bond tokens from delegator bond
-	if bond.Shares.LT(tx.Shares) return // bond shares < tx shares
+	if bond.Shares < tx.Shares return 
 	
-	bond.Shares = bond.Shares.Sub(ts.Shares)
+	bond.Shares -= tx.Shares
 
 	candidate = loadCandidate(store, tx.PubKey)
-	if candidate == nil return
-
+	
 	revokeCandidacy = false
-	if bond.Shares.IsZero() {
-        // if the bond is the owner of the candidate then trigger a revoke candidacy
-		if sender.Equals(candidate.Owner) and candidate.Status != Revoked then
+	if bond.Shares.IsZero() then
+		if sender == candidate.Owner and candidate.Status != Revoked then
 			revokeCandidacy = true
-
-		// remove the bond
-		removeDelegatorBond(store, sender, tx.PubKey)
+		removeDelegatorBond(store, sender, bond)
 	else 
 	    saveDelegatorBond(store, sender, bond)
 
-	// transfer coins back to account
 	if candidate.Status == Bonded then
         poolAccount = address of the bonded pool
     else 
         poolAccount = address of the unbonded pool
 
-	returnCoins = candidate.removeShares(shares, gs)
-	// TODO: Shouldn't it be created a queue element in this case?
-	transfer(poolAccount, sender, returnCoins)
-
+	returnedCoins = removeShares(candidate, shares)
+	
+	unbondDelegationElem = QueueElemUnbondDelegation(tx.PubKey, currentHeight(), sender, returnedCoins, startSlashRatio)
+	// TODO: Add logic regarding startSlashRatio
+	unbondDelegationQueue.add(unbondDelegationElem)
+	
+	TODO: Figure out where we should move coins after they are unbonded as they should stay somewhere during UnbondingPeriod
```

Comment:
I propose that we have a new global store named unbonding which should make everything much more clear